driverService/src/test/java/com/modsen/driverservice/controller/CarControllerTest.java

```diff
@@ -0,0 +1,385 @@
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
+                List.of(carBrandMandatory,carColorMandatory,carNumberMandatory, carModelMandatory));
+
+        MvcResult mvcResult = mockMvc.perform(put(URL + "/{carId}", carId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(emptyCarRequestDto)))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+        ListErrorMessage actualErrorResponse = objectMapper.readValue(actualResponseBody, ListErrorMessage.class);
+
+        assertThat(actualErrorResponse.errorCode()).isEqualTo(expectedErrorResponse.errorCode());
+        assertThat(actualErrorResponse.errorMessages()).containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
+    }
+
+    @Test
+    void updateCar_whenInvalidValue_thenReturns400AndErrorResult() throws Exception {
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
+                List.of(carBrandInvalid, carColorInvalid, carModelInvalid, carNumberInvalid, driverIdInvalid));
+
+        MvcResult mvcResult = mockMvc.perform(put(URL + "/{carId}", carId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(invalidCarRequestDto)))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+        ListErrorMessage actualErrorResponse = objectMapper.readValue(actualResponseBody, ListErrorMessage.class);
+
+        assertThat(actualErrorResponse.errorCode()).isEqualTo(expectedErrorResponse.errorCode());
+        assertThat(actualErrorResponse.errorMessages()).containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
+    }
+
+    @Test
+    void updateDriver_whenInvalidId_thenReturns400AndErrorResult() throws Exception {
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
+                List.of(carIdInvalid));
+
+        MvcResult mvcResult = mockMvc.perform(put(URL + "/{carId}", invalidCarId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(carRequestDto)))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+        ListErrorMessage actualErrorResponse = objectMapper.readValue(actualResponseBody, ListErrorMessage.class);
+
+        assertThat(actualErrorResponse.errorCode()).isEqualTo(expectedErrorResponse.errorCode());
+        assertThat(actualErrorResponse.errorMessages()).containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
```

Comment:
Более читабельно, если перенести на новую строку:
```suggestion
        assertThat(actualErrorResponse.errorCode())
                .isEqualTo(expectedErrorResponse.errorCode());
        assertThat(actualErrorResponse.errorMessages())
                .containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
```